cluster-api/src/main/java/io/aiven/klaw/clusterapi/utils/ClusterApiUtils.java

```diff
@@ -0,0 +1,426 @@
+    } catch (Exception e) {
+      adminClientsMap.remove(adminClientKey);
+      adminClient.close();
+      log.error("Cannot create Admin Client {} {} {}", envHost, protocol, clusterIdentification);
+      throw new Exception("Cannot connect to cluster. Please contact Administrator.");
+    }
+  }
+
+  public Properties getPlainProperties(String environment) {
+    Properties props = new Properties();
+
+    props.put("bootstrap.servers", environment);
+    setOtherConfig(props);
+
+    return props;
+  }
+
+  public Properties getSslProperties(String environment, String clusterIdentification) {
+    Properties props = getSslConfig(clusterIdentification);
+
+    props.put("bootstrap.servers", environment);
+    props.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, "SSL");
+    props.put(AdminClientConfig.CLIENT_ID_CONFIG, "klawclientssl");
+    setOtherConfig(props);
+
+    return props;
+  }
+
+  public Properties getSaslPlainProperties(String environment, String clusterIdentification) {
+    Properties props = new Properties();
+
+    props.put("bootstrap.servers", environment);
+
+    try {
+      props.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, "SASL_PLAINTEXT");
+      props.put(AdminClientConfig.CLIENT_ID_CONFIG, "klawclientsaslplain");
+      setOtherConfig(props);
+
+      if (!Strings.isNullOrEmpty(env.getProperty("kafkasasl.saslmechanism.plain"))) {
+        props.put(SaslConfigs.SASL_MECHANISM, env.getProperty("kafkasasl.saslmechanism.plain"));
+      }
+
+      if (!Strings.isNullOrEmpty(
+          env.getProperty(clusterIdentification.toLowerCase() + ".kafkasasl.jaasconfig.plain"))) {
+        props.put(
+            SaslConfigs.SASL_JAAS_CONFIG,
+            env.getProperty(clusterIdentification.toLowerCase() + ".kafkasasl.jaasconfig.plain"));
+      }
+
+    } catch (Exception exception) {
+      log.error("Error : Cannot set SASL PLAIN Config properties.");
+    }
+
+    return props;
+  }
+
+  public Properties getSaslSsl_PlainMechanismProperties(
+      String environment, String clusterIdentification) {
+    Properties props = getSslConfig(clusterIdentification);
+
+    try {
+      props.put("bootstrap.servers", environment);
+      props.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, "SASL_SSL");
+      props.put(AdminClientConfig.CLIENT_ID_CONFIG, "klawclientsaslsslplain");
+      setOtherConfig(props);
+
+      if (!Strings.isNullOrEmpty(env.getProperty("kafkasasl.saslmechanism.plain"))) {
+        props.put(SaslConfigs.SASL_MECHANISM, env.getProperty("kafkasasl.saslmechanism.plain"));
+      }
+      if (!Strings.isNullOrEmpty(
+          env.getProperty(clusterIdentification.toLowerCase() + ".kafkasasl.jaasconfig.plain"))) {
+        props.put(
+            SaslConfigs.SASL_JAAS_CONFIG,
+            env.getProperty(clusterIdentification.toLowerCase() + ".kafkasasl.jaasconfig.plain"));
+      }
+    } catch (Exception exception) {
+      log.error("Error : Cannot set SASL SSL PLAIN Config properties.");
```

Comment:
```suggestion
      log.error("Error : Cannot set SASL SSL PLAIN Config properties.", exception);
```
logging stack trace here could be also helpful